Demo/TableViewController.swift

```diff
@@ -0,0 +1,36 @@
+import UIKit
+
+class TableCell: UITableViewCell {
+
+}
+
+class TableViewController: SweetTableViewController {
+    lazy var data: [String] = {
+        return ["Some", "Stringing", "Going", "On"]
+    }()
+
+    override init(with tableViewStyle: UITableViewStyle = .Plain) {
+        super.init(with: tableViewStyle)
+
+        self.tableView.register(TableCell)
+
+        self.tableView.dataSource = self
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        fatalError("!")
+    }
+}
+
+extension TableViewController: UITableViewDataSource {
+    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+        return self.data.count
+    }
+
+    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCellWithIdentifier(TableCell.reuseIdentifier, forIndexPath: indexPath) as! TableCell
```

Comment:
We could steal/borrow this from here: http://alisoftware.github.io/swift/generics/2016/01/06/generic-tableviewcells/

```swift
func dequeueReusableCell<T: Reusable>(indexPath indexPath: NSIndexPath) -> T {
  return self.dequeueReusableCellWithIdentifier(T.reuseIdentifier, forIndexPath: indexPath) as! T
}
```